ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -165,15 +177,17 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
 
         // Fallback is unlikely to set unless echarts theme palette in charts theme provider is undefined.
         const fallbackColor =
-          Array.isArray(echartsPalette) && echartsPalette[0]
-            ? (echartsPalette[0] as string)
+          Array.isArray(categoricalPalette) && categoricalPalette[0]
+            ? (categoricalPalette[0] as string) // needed since echarts color property isn't always an array
             : muiTheme.palette.primary.main;
-        // Check which color palette was chosen and get appropriate color.
+
+        // Decide which palette to use based on total series returned.
+        // When series number exceeds number of colors in palette, use generative colors instead.
+        const paletteLength = Array.isArray(categoricalPalette) ? categoricalPalette.length : 4;
```

Comment:
is there a reason why we pick 4 here?

Reply:
Changed to `0` instead so it will call getAutoPaletteColor in the event that the categorical [palette](https://github.com/perses/perses/blob/e3cb920002695309604d36d4ca62a233dd506f8e/ui/components/src/utils/theme-gen.ts#L43-L51) in the ECharts theme isn't set.